tests/abi/testabi_key.c

```diff
@@ -140,30 +108,30 @@ static void test_keyNewSystem (void)
 	printf ("Test system key creation\n");
 
 	// Empty key
-	key = keyNew (0);
```

Comment:
What happens with the old syntax? (I am all for getting rid of it but as long it compiles we somehow need to do something with it.)

Reply:
What do you mean "the old syntax"?

`NULL` is not a valid key name anymore and is treated like any other invalid key name. If you pass an invalid key name to `keyNew`, then `keyNew` will return `NULL`.

This was always the case according to the documentation of `keyNew` on master:
https://github.com/ElektraInitiative/libelektra/blob/6d23610802c772957732826cdc87b1d4e4aeb59b/src/libs/elektra/key.c#L163

Although the implementation of `keyVNew` totally ignores the return value of `elektraKeySetName` on master and therefore doesn't agree with the documentation.

In any case, the new `keyVNew` does check the return value and does properly `free` the `Key` and return `NULL`, if the key name wasn't valid. It also logs a warning stating that the key name was invalid.

https://github.com/ElektraInitiative/libelektra/blob/14297658bee7181cfb59f08c542f2b6ccf0d02b8/src/libs/elektra/key.c#L314-L319

Reply:
> -    key = keyNew (0);
>
> What do you mean "the old syntax"?

`keyNew (0);` is the old syntax for `keyNew ("", KEY_END);`

> |NULL| is not a valid key name anymore and is treated like any other
> invalid key name. If you pass an invalid key name to |keyNew|, then
> |keyNew| will return |NULL|.

Yes, I was missing a testcase for that.

> In any case, the new |keyVNew| does check the return value and does
> properly |free| the |Key| and return |NULL|, if the key name wasn't
> valid. It also logs a warning stating that the key name was invalid.

Thank you for fixing this bug :sparkling_heart: